core/src/components/datetime/utils/parse.ts

```diff
@@ -115,6 +115,7 @@ export function parseDate(val: string | string[] | undefined | null): DatetimePa
     hour: parse[4],
     minute: parse[5],
     tzOffset,
+    ampm: parse[4] < 12 ? 'am' : 'pm',
```

Comment:
`parseDate` now includes the `ampm` information which is required for this fix.

 Do we still need this code and its implementation? https://github.com/ionic-team/ionic-framework/blob/02756c5fe408fc9957c25a1be008d5b8b708953d/core/src/components/datetime/datetime.tsx#L1210

Reply:
`parseDate` will be called on the value(s), but will not be called on the `minParts` or `maxParts`, which can be returned from `clampDate`. 

If the value is before the min or greater than the max, I believe it would result in `ampm` being `undefined` if the `parseAmPm` was removed here.

Reply:
Ah good point. I suppose we could also update `parseMinParts` and `parseMaxParts` to set the `ampm` value (https://github.com/ionic-team/ionic-framework/blob/02756c5fe408fc9957c25a1be008d5b8b708953d/core/src/components/datetime/utils/parse.ts#L185), but that seems out of scope for this PR, so I will leave `parseAmPm` alone for now.